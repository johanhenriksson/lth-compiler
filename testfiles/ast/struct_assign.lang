struct st {
    int a;
    st2 b;
}

struct st2 {
    int c;
}

int main() {
    st a;
    st2 b;
    a.a = 5;
    a.b = b;
    a.b.c = 3;

    return 0;
}
