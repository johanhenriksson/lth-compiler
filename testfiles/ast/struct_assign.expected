Program 
  List 3 item(s)
    Struct st
      TypeDecl st
      List 2 item(s)
        Field 
          IdDecl int a
            TypeId int
        Field 
          IdDecl st2 b
            TypeId st2
    Struct st2
      TypeDecl st2
      List 1 item(s)
        Field 
          IdDecl int c
            TypeId int
    Function main()
      IdDecl int main
        TypeId int
      List 0 item(s)
      Block body
        List 6 item(s)
          DeclareStmt st a;
            IdDecl st a
              TypeId st
            Opt ?
          DeclareStmt st2 b;
            IdDecl st2 b
              TypeId st2
            Opt ?
          AssignStmt a = 5;
            StructIdUse a.a
              IdUse a
            IntLiteral 5
          AssignStmt b = b;
            StructIdUse a.b
              IdUse a
            IdExpr b
              IdUse b
          AssignStmt c = 3;
            StructIdUse a.b.c
              StructIdUse a.b
                IdUse a
            IntLiteral 3
          ReturnStmt return 0;
            Opt ?
              IntLiteral 0
